### PR TITLE
Revision update

### PR DESCRIPTION
**Pull request by johli**
### Description of your changes

- Added new sh script to automatically install environment variables.
- Re-added the old/original gene.output_slice.
- Added name attribute to gene class.
- Added logic for optionally using old data transforms in hound_data.
- Cleaned up seqnn.gradients function.
- Added missing stream class.
- Updated README with install instructions.

### Type of change
- [X] Bug fix
- [X] New feature
- [X] Refactoring / code clean-up
- [X] Documentation add / update

### (If applicable) How has this been tested?

Tests:
- Re-ran all scripts from 'borzoi-paper' repo and 'borzoi' tutorials.
- Checked that all major results from the paper reproduced (test evals, e/s/paQTLs).

Environment:
- Python 3.10
- TensorFlow 2.15

### CODE REVIEW

File: env_vars.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+if ! [ -e $file_vars_deact ]; then
+    echo '#!/bin/sh' > $file_vars_deact
+fi
+
+# append env variable exports to /activate.d/env_vars.sh
+echo "export BASKERVILLE_DIR=$LOCAL_BASKERVILLE_PATH" >> $file_vars_act
+echo 'export PATH=$BASKERVILLE_DIR/src/baskerville/scripts:$PATH' >> $file_vars_act
+echo 'export PYTHONPATH=$BASKERVILLE_DIR/src/baskerville/scripts:$PYTHONPATH' >> $file_vars_act
+
+echo "export BASKERVILLE_CONDA=/home/$LOCAL_USER/anaconda3/etc/profile.d/conda.sh" >> $file_vars_act
```
Comment (davek44) [inline env_vars.sh:26]: Maybe just ask for this path directly and let go the LOCAL_USER variable. I don't know how other people's systems will work for this.

File: src/baskerville/stream.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+    def fetch_batch(self):
+        """Fetch a batch of data from the dataset iterator."""
+        x = [next(self.dataset_iter)]
+        while x[-1] and len(x) < self.stream_seqs:
+            x.append(next(self.dataset_iter))
+        return x
+
+
+class PredStreamSonnet:
```
Comment (davek44) [inline src/baskerville/stream.py:115]: This is for Enformer, so you probably don't need it. Check which classes you need and delete the rest.

Comment (johli) [reply src/baskerville/stream.py:115]: Good point, I'm removing everything but PredStreamGen.

File: env_vars.sh (hunk 1 above)
Comment (johli) [reply env_vars.sh:26]: Will do!

**Comment by johli** [comment]
All requested changes are made.

**Pull request merged**